### ISSUE TITLE
App crash using BNM 1.4

### ISSUE THREAD

**Issue by nik2143** [ByNameModding/BNM-Android#19]
Game tested: Subway Surfers
Crash Log
```
12-09 17:08:24.860 31517 31517 I crash_dump32: performing dump of process 31449 (target tid = 31449)
12-09 17:08:24.863 31517 31517 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
12-09 17:08:24.863 31517 31517 F DEBUG   : Build fingerprint: 'samsung/beyond1qlteue/beyond1q:9/LMY48Z/901230529:user/release-keys'
12-09 17:08:24.863 31517 31517 F DEBUG   : Revision: '0'
12-09 17:08:24.863 31517 31517 F DEBUG   : ABI: 'x86'
12-09 17:08:24.863 31517 31517 F DEBUG   : pid: 31449, tid: 31449, name: iloo.subwaysurf  >>> com.kiloo.subwaysurf <<<
12-09 17:08:24.863 31517 31517 F DEBUG   : signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x4
12-09 17:08:24.863 31517 31517 F DEBUG   : Cause: null pointer dereference
12-09 17:08:24.863 31517 31517 F DEBUG   :     eax c9554040  ebx 80000009  ecx 00000001  edx f2df3494
12-09 17:08:24.863 31517 31517 F DEBUG   :     edi 00000004  esi e7987580
12-09 17:08:24.863 31517 31517 F DEBUG   :     ebp ea914760  esp ffd41e90  eip ea2f86a5
12-09 17:08:24.863 31517 31517 F DEBUG   :
12-09 17:08:24.863 31517 31517 F DEBUG   : backtrace:
12-09 17:08:24.863 31517 31517 F DEBUG   :     #00 pc 001366a5  /system/vendor/lib/libhoudini.so
12-09 17:08:25.130  1643  1643 E /system/bin/tombstoned: Tombstone written to: /data/tombstones/tombstone_13
12-09 17:08:25.136  1801 31533 W ActivityManager:   Force finishing activity com.kiloo.subwaysurf/com.sybogames.chili.multidex.ChiliMultidexSupportActivity
12-09 17:08:25.139  1801 31533 W ActivityManager: Exception thrown during pause
12-09 17:08:25.139  1801 31533 W ActivityManager: android.os.DeadObjectException
12-09 17:08:25.139  1801 31533 W ActivityManager:       at android.os.BinderProxy.transactNative(Native Method)
12-09 17:08:25.139  1801 31533 W ActivityManager:       at android.os.BinderProxy.transact(Binder.java:1129)
12-09 17:08:25.139  1801 31533 W ActivityManager:       at android.app.IApplicationThread$Stub$Proxy.scheduleTransaction(IApplicationThread.java:1777)
12-09 17:08:25.139  1801 31533 W ActivityManager:       at android.app.servertransaction.ClientTransaction.schedule(ClientTransaction.java:129)
12-09 17:08:25.139  1801 31533 W ActivityManager:       at com.android.server.am.ClientLifecycleManager.scheduleTransaction(ClientLifecycleManager.java:47)
12-09 17:08:25.139  1801 31533 W ActivityManager:       at com.android.server.am.ClientLifecycleManager.scheduleTransaction(ClientLifecycleManager.java:69)
12-09 17:08:25.139  1801 31533 W ActivityManager:       at com.android.server.am.ActivityStack.startPausingLocked(ActivityStack.java:1463)
12-09 17:08:25.139  1801 31533 W ActivityManager:       at com.android.server.am.ActivityStack.finishActivityLocked(ActivityStack.java:3738)
12-09 17:08:25.139  1801 31533 W ActivityManager:       at com.android.server.am.ActivityStack.finishActivityLocked(ActivityStack.java:3680)
12-09 17:08:25.139  1801 31533 W ActivityManager:       at com.android.server.am.ActivityStack.finishTopCrashedActivityLocked(ActivityStack.java:3563)
12-09 17:08:25.139  1801 31533 W ActivityManager:       at com.android.server.am.ActivityStackSupervisor.finishTopCrashedActivitiesLocked(ActivityStackSupervisor.java:2261)
12-09 17:08:25.139  1801 31533 W ActivityManager:       at com.android.server.am.AppErrors.handleAppCrashLocked(AppErrors.java:747)
12-09 17:08:25.139  1801 31533 W ActivityManager:       at com.android.server.am.AppErrors.makeAppCrashingLocked(AppErrors.java:579)
12-09 17:08:25.139  1801 31533 W ActivityManager:       at com.android.server.am.AppErrors.crashApplicationInner(AppErrors.java:447)
12-09 17:08:25.139  1801 31533 W ActivityManager:       at com.android.server.am.AppErrors.crashApplication(AppErrors.java:392)
12-09 17:08:25.139  1801 31533 W ActivityManager:       at com.android.server.am.ActivityManagerService.handleApplicationCrashInner(ActivityManagerService.java:15468)
12-09 17:08:25.139  1801 31533 W ActivityManager:       at com.android.server.am.NativeCrashListener$NativeCrashReporter.run(NativeCrashListener.java:85)
```

**Comment by BNM-Dev** [comment]
I see that crash happens on emulator, but I don't see any other useful information. You just add BNM v1.4 to project?

**Comment by BNM-Dev** [comment]
You loading it using BNM::TryForceLoadIl2CppByPath?

**Comment by nik2143** [comment]
Still crash

**Comment by BNM-Dev** [comment]
It crash even with `BNM_DISABLE_AUTO_LOAD` defined and without loading BNM?

**Comment by nik2143** [comment]
Yes crash even with BNM_DISABLE_AUTO_LOAD

**Comment by BNM-Dev** [comment]
So idk, BNM can't crash without doing at least something. And how I tested, it work emulators

**Comment by nik2143** [comment]
I put the on loaded event idk If this can make BNM do Something

**Comment by BNM-Dev** [comment]
Only auto loading, `BNM::TryForceLoadIl2CppByPath` or any `BNM::External` method, can make BNM start working. Try remove `AddOnLoadedEvent` call

**Comment by nik2143** [comment]
Removing the call to AddOnLoadedEvent solved the crash but obv don't load anything

**Comment by BNM-Dev** [comment]
Which emulator you use?

**Comment by nik2143** [comment]
Nox Emulator

**Comment by Creator1A** [comment]
> Removing the call to AddOnLoadedEvent solved the crash but obv don't load anything

I am facing the same issue. I had to shift back to using the thread load method. This method works but crashes the game when opening and it's practically just luck for the game to work.

**Comment by BNM-Dev** [comment]
> > Removing the call to AddOnLoadedEvent solved the crash but obv don't load anything
> 
> I am facing the same issue. I had to shift back to using the thread load method. This method works but crashes the game when opening and it's practically just luck for the game to work. Tested on 3 devices, S23U, Redemi, and Bluestacks/Nox

This is strange, because you can try this mod [youtube](https://youtu.be/TI9xgpvhhWM). It work on all emulators and use BNM 1.4_dev with `BNM::TryForceLoadIl2CppByPath` and `BNM::AddOnLoadedEvent` (only some fixes, not related to loading). I will release v1.5 soon, maybe it will fix issue.

**Comment by Creator1A** [comment]
Commenting out the lines related to events in the BNM.cpp and using thread detach seem to have fixed the crash for me. Using AddOnLoadedEvent still crashes the game no matter what for some reason.

**Comment by BNM-Dev** [comment]
`OnLoadedEvents` are fixed in v1.6.
